firmware/projects/FrontController/inc/simp_vd_interface.h

```diff
@@ -0,0 +1,44 @@
+/// @author Teghveer Singh Ateliey
+/// @date 2024-11-23
+
+#pragma once
+
+#include "app.h"
+#include "shared/controls/motor_torque.h"
+#include "shared/controls/tc_scale_factor.h"
+#include "shared/controls/tvFactor.h"
+
+struct VdInput {
+    float driver_torque_request;
+    float brake_pedal_postion;
+    float steering_angle;
+    float wheel_speed_lr;
+    float wheel_speed_rr;
+    float wheel_speed_lf;
+    float wheel_speed_rf;
+    bool tv_enable;
+};
+
+struct VdOutput {
+    float lm_torque_limit_positive;
+    float rm_torque_limit_positive;
+    float lm_torque_limit_negative;
+    float rm_torque_limit_negative;
+    uint16_t left_motor_speed_request;
+    uint16_t right_motor_speed_request;
+};
+
+class SimpVdInterface {
+public:
+    SimpVdInterface(
+        float target_slip = 0.2f);  // default target slip is float 0.2
+    VdOutput update(const VdInput& input, int time_ms);
+
+private:
+    float target_slip;
+    float motor_torque_request;
```

Comment:
why are these class variables? If possible, make them local to the update function